lablup/backend.ai-webui
# feat: E2E test using Playwright

This PR resolves https://github.com/lablup/backend.ai-webui/issues/2081

Implement e2e test using playwright
[Loop Doc](https://lablup.sharepoint.com/:fl:/s/fb28b4ec-f199-4e97-a2b6-c73a1fcdd1ff/ETnciCtsG5ZKmBN4dxi2_2gBAvbgtYNmqRoklJoqKSIylw?e=obSCqL&nav=cz0lMkZzaXRlcyUyRmZiMjhiNGVjLWYxOTktNGU5Ny1hMmI2LWM3M2ExZmNkZDFmZiZkPWIlMjFlS3Z6RE5UdHUwbWxLRVJwTEhPeWtibHFzYU12SDlkS2xLeTZHa29tcVUyUTBZcTdyUFJDVEpZdkxqVktWU2h3JmY9MDFXUkVWVkRKWjNTRUNXM0EzU1pGSlFFM1lPNE1MTjczSSZjPSUyRiZhPUxvb3BBcHAmcD0lNDBmbHVpZHglMkZsb29wLXBhZ2UtY29udGFpbmVyJng9JTdCJTIydyUyMiUzQSUyMlQwUlRVSHhzWVdKc2RYQXVjMmhoY21Wd2IybHVkQzVqYjIxOFlpRmxTM1o2UkU1VWRIVXdiV3hMUlZKd1RFaFBlV3RpYkhGellVMTJTRGxrUzJ4TGVUWkhhMjl0Y1ZVeVVUQlpjVGR5VUZKRFZFcFpka3hxVmt0V1UyaDNmREF4VjFKRlZsWkVTVmd5VEVaSVRVNWFTelpPUkVsTVUwVk1XRk5DUmtKTVFqVSUzRCUyMiUyQyUyMmklMjIlM0ElMjI3ZjMyMTZhMy1hZWJjLTRmZGEtYjkxMi1lYWRjNGIxMzFhYWYlMjIlN0Q%3D)

You must make .env file at playwright directory(webui/playwright/.env) and copy and paste below config
```
SUPER_ADMIN_EMAIL=admin@lablup.com
SUPER_ADMIN_PASSWORD=wJalrXUt
ADMIN_EMAIL=domain-admin@lablup.com
ADMIN_PASSWORD=cWbsM_vB
USER_EMAIL=user@lablup.com
USER_PASSWORD=C8qnIo29
USER2_EMAIL=user2@lablup.com
USER2_PASSWORD=P7oxTDdz
INCORRECT_EMAIL=test7@lablup.com
INCORRECT_PASSWORD=dasdnsajdnk
PAGE_URL=http://localhost:9081
ENDPOINT=http://localhost:8090
CREATE_VFOLDER_GROUP=test_peoject
NOT_CREATE_VFOLDER_GROUP=default
DELETE_VFOLDER=vMyFkaomKN9vkTORaywaleDayTjYGXwDyB4LPxUiLFvVAWCW2iFjEtJzzXZ8gNK3
NEW_SESSION_VERSION=Ubuntu 20.04 aarch64
NEW_SESSION_MOUNTED_VFOLDER=test
DELETE_SESSION=vNkj
CREATE_USER_NAME=test9
CREATE_USER_EMAIL=test9@lablup.com
CREATE_USER_PASSWORD=test123!
DELETE_USER_EMAIL=test9@lablup.com
```
How to run
1. Follow [extension guide](https://playwright.dev/docs/getting-started-vscode#running-tests) to install extension
2. Pull this branch
3. Open test explorer at left side bar as in the picture below
<img width="1552" alt="스크린샷 2023-12-26 오후 6 03 33" src="https://github.com/lablup/backend.ai-webui/assets/28584221/6c4ecf5e-f244-41b3-9a6b-1901112ded0b">
4. Click to play button at test explorer or beside of code lines to run test 
<img width="1437" alt="스크린샷 2023-12-26 오후 6 06 55" src="https://github.com/lablup/backend.ai-webui/assets/28584221/34e68e2b-e69e-4c61-9de8-2f3a34c28aec">
5. You can check the testing process through browser and test result on terminal
<img width="1552" alt="스크린샷 2023-12-26 오후 6 13 36" src="https://github.com/lablup/backend.ai-webui/assets/28584221/36ec291d-89a0-4544-b977-818867bea0fa">
<img width="1122" alt="스크린샷 2023-12-26 오후 6 13 55" src="https://github.com/lablup/backend.ai-webui/assets/28584221/6cb86e8d-6faf-42c5-9f7b-be472de5531c">



**Checklist:** (if applicable)
